daemon/fqdn.go

```diff
@@ -100,6 +101,17 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 		fqdn.StartDNSPoller(d.dnsPoller)
 	}
 
+	// Controller to cleanup TTL expired entries from the DNS policies.
+	controller.NewManager().UpdateController("dns-ttl-cleanup-job", controller.ControllerParams{
+		RunInterval: time.Second,
```

Comment:
I'm probably the wrong person to review this PR but do we need to run this controller every second? Can't this be event based on the TTL of the entry?

Reply:
no, each second the cleanup need to happens to clean the table.

Reply:
But the entries don't have a TTL for every second

Reply:
Multiple dns queries has multiple DNS TTL, so each second we need to clean the TTL that has expired after X seconds.

Reply:
The challenge is knowing when to wake up to expire a specific IP and regenerate policy. We know the expiration time of each entry and setting a timer for the next one to expire is straight-forward Figuring out the expire time for the one after the next is annoying. We would need something like a priority queue (with O(logn) best case update). In theory, most updates will add a new time to the end of this queue (i.e. a new latest time). Still, we can have situations where new entries will expire somewhere in the middle.
Binning the entries into seconds seems like the lowest overhead, since we have as many map entries as we have IPs (or less). The lookup is also constant time for a specific second.

Reply:
I doubt that we need 1 second accuracy for the TTL. Isn't it enough to expire every 15 seconds?